apps/emqx_durable_storage/src/emqx_ds_storage_bitfield_lts.erl

```diff
@@ -556,6 +568,23 @@ delete_next_until(
         rocksdb:iterator_close(ITHandle)
     end.
 
+-spec lookup_message(emqx_ds_storage_layer:shard_id(), s(), emqx_ds_precondition:matcher()) ->
+    emqx_types:message() | not_found | emqx_ds:error(_).
+lookup_message(
+    _ShardId,
+    S = #s{db = DB, data = CF},
+    #message_matcher{topic = Topic, timestamp = Timestamp}
```

Comment:
Q: this matches a message with _exactly_ the same timestamp as specified in the matcher, not _the first message at or after_ the supplied timestamp, right?

Reply:
Yes. Idea is that the timestamp is known, because it's (most of the time) part of the message key, either through static assigment or `emqx_ds:next/3`. The latter usecase wouldn't have worked before actually, so I changed the approach to timestamps in 58b9ab021088829e2b7444a5b0ad3e9c42fddf0b slightly.